pkg/devserver/devserver.go

```diff
@@ -182,12 +184,12 @@ func start(ctx context.Context, opts StartOpts) error {
 			backoff.GetLinearBackoffFunc(time.Duration(opts.RetryInterval)*time.Second),
 		))
 	}
-	queue := redis_state.NewQueue(rc, queueOpts...)
+	queue := redis_state.NewQueue(unshardedClient.Queue(), queueOpts...)
```

Comment:
Each "feature" now receives its own client, including a Redis connection and key generator.